### ISSUE TITLE
edgedb error: InternalServerError: out of shared memory

### ISSUE THREAD

**Issue by monomonedula** [edgedb/edgedb#5018]
Suddenly started getting `edgedb error: InternalServerError: out of shared memory` when creating (and migrating) a new db.
```
Applied m1pppmafsgqmdkmk7dfx4dtsrtgju3gecp7qdan3aqrwub4kzdsqhq (00001.edgeql)
Applied m1uqulra2jqc6msweshn5z6loqfrrtf53r3u4fag7yoznu6oqpsdya (00002.edgeql)
Applied m1erspek6pmsucjs4ap4ctubkwohvaydg66spmczo437cbm4l37mvq (00003.edgeql)
edgedb error: InternalServerError: out of shared memory
  Hint: This is most likely a bug in EdgeDB. Please consider opening an issue ticket at https://github.com/edgedb/edgedb/issues/new?template=bug_report.md
  Server traceback:
      edb.errors.InternalServerError: out of shared memory
```

- EdgeDB Version: `2.9+a9c5c6f`
- EdgeDB CLI Version: `EdgeDB CLI 2.2.6+b869667`
- OS Version: `Ubuntu 22.04.1 LTS`

Steps to Reproduce:

I don't know actually, it all worked OK and all of sudden it stopped.
I just create a new DB and try to apply migrations and this happens.

It's not about the migrations themselves because 
1) It worked fine before
2) It sometimes fails before getting to the migration part
3) It fails on different migrations

**Comment by monomonedula** [comment]
If it makes any difference, the server is used for testing so new databases are created/deleted quite often

**Comment by elprans** [comment]
Yes, this is `max_locks_per_transaction`, but that should only depend on the number of _tables_, not data, so deleting the data could have been a coincidence?  When you say "a lot of links", does it refer to the number of link _declarations_ or actual link records?

**Comment by monomonedula** [comment]
@elprans it doesn't look like a coincidence since it started to just fine exactly after i deleted the data. By a lot of links I mean that the schema consists of 42 types and 46 links among them (some of them one-to-one, some of them one-to-many, some of them backlinks). There are also some computable fields.

**Comment by monomonedula** [comment]
@elprans aaand i'm getting this error again after the DB reached about the same threshold.

**Comment by monomonedula** [comment]
Still desperate for help here
